tests/test_rerender_crystals.py

```
@@ -0,0 +1,264 @@
+        contra = (vault / CRYSTAL_DIR_REL / "contradiction-ren01.md").read_text(
+            encoding="utf-8",
+        )
+        assert "## 相关笔记" in contra
+        assert "[[a]]" in contra and "[[b]]" in contra
+        assert "## 争议核心" in contra
+
+        out = capsys.readouterr().out
+        assert "community crystals rewrote:" in out
+        assert "contradiction crystals rewrote:" in out
+
+    def test_dry_run_does_not_touch_files(self, tmp_path, capsys):
```

Ruff (0.15.12)

[warning] Unused method argument: `capsys`

(ARG002)